packages/backend/src/core/SequenceProcessor.ts

```diff
@@ -117,9 +121,14 @@ export class SequenceProcessor extends EventEmitter {
   private async process(): Promise<void> {
     this.logger.debug('Processing started')
 
+    let firstRun = true
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
     processing: while (true) {
-      const lastProcessed = this.state?.lastProcessed
+      let lastProcessed = this.state?.lastProcessed
+      // we need to adjust starting block if its first run and uncertaintyBuffer is set
+      if (firstRun && lastProcessed && this.uncertaintyBuffer > 0) {
```

Comment:
This implicitly relies on heroku's behavior of restarting the process every day. Maybe if uncertainty buffer is set we should always set `lastProcessed = actualLastProcessed - uncertaintyBuffer`? Or only if we're on the tip?

Reply:
I am not sure if I understand, `process()` gets called on hourly basis so it should work fine. Please ping me if I am wrong but I am gonna merge it now.